src/main/scala/com/gu/acquisitionFirehoseTransformer/Lambda.scala

```diff
@@ -0,0 +1,41 @@
+package com.gu.acquisitionFirehoseTransformer
+
+import java.nio.ByteBuffer
+import collection.JavaConverters._
+import com.typesafe.scalalogging.LazyLogging
+
+import com.amazonaws.services.lambda.runtime.events.KinesisAnalyticsInputPreprocessingResponse._
+import com.amazonaws.services.lambda.runtime.events.{KinesisAnalyticsInputPreprocessingResponse, KinesisFirehoseEvent}
+import com.amazonaws.services.lambda.runtime.Context
+
+import com.gu.thrift.serializer.ThriftDeserializer
+import ophan.thrift.event.Acquisition
+
+object Lambda extends LazyLogging {
+
+  def handler(event: KinesisFirehoseEvent, context: Context): KinesisAnalyticsInputPreprocessingResponse = {
+    val records = event.getRecords.asScala
+
+    val results = records.flatMap { record =>
+      ThriftDeserializer.deserialize(record.getData.array)(Acquisition).fold(
+        (e: Throwable) => {
+          logger.error(s"Error deserializing record: ${e.getMessage}")
+          None
+        },
+        (acquisition: Acquisition) => {
+          val csvRow = AcquisitionToCsv(acquisition, record.getApproximateArrivalTimestamp)
+
+          Some {
+            new Record(record.getRecordId, Result.Ok, ByteBuffer.wrap(csvRow.getBytes))
+          }
+        }
+      )
+    }
+
+    logger.info(s"Sending ${results.length} csv rows from ${records.length} input records.")
+
+    val response = new KinesisAnalyticsInputPreprocessingResponse()
```

Comment:
I'm interested in how this maps to individual objects in S3, and where that is controlled (I suspect it's the buffering settings [here](https://github.com/guardian/contributions-platform/pull/123/files) - any comments to that effect would be useful!

Reply:
yes the buffer settings can be configured in that cloudformation template. You can configure this buffering separately for both the lambda invocation and s3 writes.
I can document this in contributions-platform